cranelift/codegen/src/isa/s390x/abi.rs

```diff
@@ -327,40 +327,44 @@ impl ABIMachineSpec for S390xMachineDeps {
             let vecreg = in_vec_reg(param.value_type);
             debug_assert!(intreg as i32 + fltreg as i32 + vecreg as i32 <= 1);
 
-            let (next_reg, candidate, implicit_ref) = if intreg {
+            let (candidate, implicit_ref) = if intreg {
                 let candidate = match args_or_rets {
                     ArgsOrRets::Args => get_intreg_for_arg(call_conv, next_gpr),
                     ArgsOrRets::Rets => get_intreg_for_ret(call_conv, next_gpr),
                 };
-                (&mut next_gpr, candidate, None)
+                next_gpr += 1;
```

Comment:
Not sure I understand the point of this change.  This now increments next_gpr *always*, even when get_intreg_for_arg returns None, i.e. no GPR is actually used.  This is only correct under the implicit assumption that if get_intreg_for_arg failed once, it will always fail in the future.  That may be true with the current ABI, but not necessarily with any possible future ABIs ...